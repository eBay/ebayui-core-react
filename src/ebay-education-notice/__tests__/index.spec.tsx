import React from 'react'
import { render } from '@testing-library/react'
import EbayIcon from '../../ebay-icon/icon'
import {
    EbayEducationNotice,
    EbayNoticeContent,
    EbayEducationNoticeTitle,
    EbayEducationNoticeFooter
} from '../index'
import userEvent from '@testing-library/user-event'
import { act } from 'react-dom/test-utils'

describe('<EbayEducationNotice>', () => {
    it('should render notice snapshot', () => {
        const { asFragment } = render(
            <EbayEducationNotice a11yText="a11y notice">
                <EbayIcon name="theEbayVault24" />
                <EbayEducationNoticeTitle>
                    Recommended title format
                </EbayEducationNoticeTitle>
                <EbayNoticeContent>
                    <p>
                        Follow the order below to optimize market valuation from
                        Price Guide. Player + Set or Season + Manufacturer +
                        Card number + Variant + Grader + Grade
                    </p>
                </EbayNoticeContent>
                <EbayEducationNoticeFooter>
                    Education footer
                </EbayEducationNoticeFooter>
            </EbayEducationNotice>
        )

        expect(asFragment()).toMatchSnapshot()
    })
    it('should render notice prominent snapshot', () => {
        const { asFragment } = render(
            <EbayEducationNotice
                variant="prominent"
                iconVariant="prominent"
                a11yDismissText="dismiss notice"
            >
                <EbayIcon name="theEbayVault24" />
                <EbayEducationNoticeTitle>
                    Recommended title format
                </EbayEducationNoticeTitle>
                <EbayNoticeContent>
                    <p>
                        Follow the order below to optimize market valuation from
                        Price Guide. Player + Set or Season + Manufacturer +
                        Card number + Variant + Grader + Grade
                    </p>
                </EbayNoticeContent>
            </EbayEducationNotice>
        )

        expect(asFragment()).toMatchSnapshot()
    })
    it('should no render without title', () => {
        expect(() => {
            render(
                <EbayEducationNotice
                    variant="prominent"
                    iconVariant="prominent"
                    a11yDismissText="dismiss notice"
                >
                    <EbayIcon name="theEbayVault24" />
                    <EbayNoticeContent>
                        <p>
                            Follow the order below to optimize market valuation
                            from Price Guide. Player + Set or Season +
                            Manufacturer + Card number + Variant + Grader +
                            Grade
                        </p>
                    </EbayNoticeContent>
                </EbayEducationNotice>
            )
        }).toThrow(
            'EbayEducationNoticeTitle: Please use a EbayEducationNoticeTitle that defines the content of the notice'
        )
    })

    it('onDismiss being fired', async () => {
        const onDismissMock = jest.fn()
        const { getByRole } = render(
            <EbayEducationNotice
                educationIcon="lightbulb24"
                a11yDismissText="notice dismiss"
                onDismiss={onDismissMock}
            >
                <EbayIcon name="theEbayVault24" />
                <EbayEducationNoticeTitle>
                    Recommended title format
                </EbayEducationNoticeTitle>
                <EbayNoticeContent>
                    <p>
                        Follow the order below to optimize market valuation from
                        Price Guide. Player + Set or Season + Manufacturer +
                        Card number + Variant + Grader + Grade
                    </p>
                </EbayNoticeContent>
            </EbayEducationNotice>
        )

        const dismissBtn = getByRole('button', { name: 'notice dismiss' })
        expect(dismissBtn).toBeInTheDocument()
        await act(async () => {
            await userEvent.click(dismissBtn)
        })

        expect(onDismissMock).toHaveBeenCalledTimes(1)
    })
})
