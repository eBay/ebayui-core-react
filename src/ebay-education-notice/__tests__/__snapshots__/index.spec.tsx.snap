// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayEducationNotice> should render notice prominent snapshot 1`] = `
<DocumentFragment>
  <section
    aria-roledescription="Notice"
    class="education-notice education-notice--prominent"
    role="region"
  >
    <div
      class="education-notice__header"
    >
      <svg
        aria-hidden="true"
        class="icon--prominent icon icon--24"
        focusable="false"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-the-ebay-vault-24"
        />
      </svg>
      <h2
        class="education-notice__title"
      >
        Recommended title format
      </h2>
      <button
        aria-label="dismiss notice"
        class="education-notice__dismiss icon-btn icon-btn--small"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-close-16"
          />
        </svg>
      </button>
    </div>
    <div
      class="education-notice__main"
    >
      <p>
        Follow the order below to optimize market valuation from Price Guide. Player + Set or Season + Manufacturer + Card number + Variant + Grader + Grade
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<EbayEducationNotice> should render notice snapshot 1`] = `
<DocumentFragment>
  <section
    aria-roledescription="Notice"
    class="education-notice"
    role="region"
  >
    <div
      class="education-notice__header"
    >
      <svg
        aria-label="a11y notice"
        class="icon icon--24"
        focusable="false"
        role="img"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-the-ebay-vault-24"
        />
      </svg>
      <h2
        class="education-notice__title"
      >
        Recommended title format
      </h2>
    </div>
    <div
      class="education-notice__main"
    >
      <p>
        Follow the order below to optimize market valuation from Price Guide. Player + Set or Season + Manufacturer + Card number + Variant + Grader + Grade
      </p>
    </div>
    <div
      class="education-notice__footer"
    >
      Education footer
    </div>
  </section>
</DocumentFragment>
`;
